src/overlays/actors/ovl_En_Mnk/z_en_mnk.c

```diff
@@ -28,191 +54,2198 @@ ActorInit En_Mnk_InitVars = {
+        this->unk_3E4 |= 4;
+    }
+
+    this->picto.actor.world.rot.y = this->picto.actor.yawTowardsPlayer;
+    Math_SmoothStepToS(&this->picto.actor.shape.rot.y, this->picto.actor.world.rot.y, 2, 0xBB8, 0xC8);
+
+    if (SkelAnime_Update(&this->skelAnime)) {
+        Animation_PlayLoop(&this->skelAnime, &object_mnk_Anim_008814);
+        this->actionFunc = func_80AB6304;
+    }
+
+    if (this->picto.actor.velocity.y < 0.0f) {
+        this->unk_3E4 &= ~2;
+    }
+
+    if (Animation_OnFrame(&this->skelAnime, 5.0f)) {
+        Actor_PlaySfx(&this->picto.actor, NA_SE_EV_MONKEY_VO_JOY);
+    }
+}
+
+void func_80AB64B8(EnMnk* this, PlayState* play) {
+    if (!(this->unk_3E4 & 0x8)) {
+        SkelAnime_Update(&this->skelAnime);
+        this->picto.actor.world.rot.y = this->picto.actor.yawTowardsPlayer;
+        Math_SmoothStepToS(&this->picto.actor.shape.rot.y, this->picto.actor.world.rot.y, 2, 0xBB8, 0xC8);
+    }
+
+    if (Actor_ProcessTalkRequest(&this->picto.actor, &play->state)) {
+        this->picto.actor.flags &= ~ACTOR_FLAG_10000;
+        this->actionFunc = func_80AB63CC;
+        EnMnk_Monkey_ChangeAnim(this, 9, ANIMMODE_ONCE, -5.0f);
+        this->picto.actor.velocity.y = 3.6f;
+        this->picto.actor.terminalVelocity = -9.0f;
+        this->picto.actor.gravity = -0.4f;
+        this->unk_3E4 &= ~1;
+        this->unk_3E4 &= ~4;
+    } else if (this->picto.actor.xzDistToPlayer < 100.0f) {
+        this->picto.actor.flags |= ACTOR_FLAG_10000;
+        Actor_OfferTalk(&this->picto.actor, play, 120.0f);
+    } else {
+        this->picto.actor.flags &= ~ACTOR_FLAG_10000;
+    }
+}
+
+void EnMnk_Monkey_RunAndWaitAtEachPoint(EnMnk* this, PlayState* play) {
+    MonkeyFollowPathResult followPathResult;
+
+    SkelAnime_Update(&this->skelAnime);
+    followPathResult =
+        EnMnk_FollowPath(this, MONKEY_FOLLOWPATH_FLAGS_REACHNEXTPOINT | MONKEY_FOLLOWPATH_FLAGS_LONGRANGE);
+    if (followPathResult != MONKEY_FOLLOWPATH_RESULT_FOLLOWING) {
+        if ((followPathResult == MONKEY_FOLLOWPATH_RESULT_SHORTRANGE_FINISHEDPATH) ||
+            (followPathResult == MONKEY_FOLLOWPATH_RESULT_LONGRANGE_FINISHEDPATH)) {
+            this->pathIndex = this->path->additionalPathIndex;
+            if (this->pathIndex == ADDITIONAL_PATH_INDEX_NONE) {
+                this->pathIndex = PATH_INDEX_NONE;
+            }
+            this->path = &play->setupPathList[this->pathIndex];
+            this->destPointIndex = 0;
+            EnMnk_Monkey_SetAnim(this, 0);
+            this->actionFunc = func_80AB64B8;
+            this->picto.actor.terminalVelocity = -9.0f;
+            this->picto.actor.velocity.y = 0.0f;
+            this->picto.actor.speed = 0.0f;
+            this->picto.actor.gravity = -1.0f;
+        } else if (this->picto.actor.xzDistToPlayer > 160.0f) {
+            this->actionFunc = EnMnk_Monkey_WaitToRunAndWaitAtEachPoint;
+            this->unk_3E4 |= 3;
```

Comment:
Same thing about unk_3E4 here:
```suggestion
            this->unk_3E4 |= (2 | 1);
```